testnet/stacks-node/src/nakamoto_node/sign_coordinator.rs

```diff
@@ -177,6 +183,16 @@ fn get_signer_commitments(
 
         return Ok(party_polynomials);
     }
+
+    let computed_key = all_party_polynomials
+        .iter()
+        .fold(Point::default(), |s, (_, comm)| s + comm.poly[0]);
+
+    if &computed_key == expected_aggregate_key {
+        debug!("Aggregate key computed from combined DKG results match expected. Using this one");
+        return Ok(all_party_polynomials.into_iter().collect());
+    }
```

Comment:
What case is this capturing? My understanding is that honest signers produce a DkgResults message that has all of the party polynomials required to compute the aggregate key. So, the only case in which DkgResults aren't recoverable via a single signer's DkgResults is if all of the signer's DkgResults messages are modified specifically to make the aggregate key unrecoverable by signers. This would be dishonest behavior on the part of all the signers, so why would they include any such information to recover the aggregate key at all (even aggregated across all of the signers)?

Reply:
So this is for the extra DkgResults message that is sent by signers when they are loading their persisted shares. In that scenario, the signer only has their own party polynomial available so the sign coordinator has to aggregate all of the DkgResults messages to get all of the party polynomials.

Again, I'm not super happy about that solution and open to suggestions - but any alternative I have considered so far would be a much more extensive change.

Reply:
Removed this case now since I updated the signer to only send the messages after an aggregate key has been approved - so the extra DkgResults message doesn't exist any more and we don't need this case.